sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -699,49 +504,73 @@ public Write withQueueDeclare(boolean queueDeclare) {
     private static class WriteFn extends DoFn<RabbitMqMessage, Void> {
       private final Write spec;
 
-      private transient ConnectionHandler connectionHandler;
+      private UUID writerId;
+      private transient ChannelLeaser channelLeaser;
 
       WriteFn(Write spec) {
         this.spec = spec;
       }
 
       @Setup
       public void setup() throws Exception {
-        connectionHandler = new ConnectionHandler(spec.uri());
-        connectionHandler.start();
-
-        Channel channel = connectionHandler.getChannel();
-
-        if (spec.exchange() != null && spec.exchangeDeclare()) {
-          channel.exchangeDeclare(spec.exchange(), spec.exchangeType());
+        if (writerId == null) {
+          writerId = UUID.randomUUID();
         }
-        if (spec.queue() != null && spec.queueDeclare()) {
-          channel.queueDeclare(spec.queue(), true, false, false, null);
+        if (channelLeaser == null) {
+          channelLeaser = spec.connectionHandlerProviderFn().apply(null);
```

Comment:
man, annoying beam doesn't have a `SerializableSupplier` or `SerializableProvider`.  passing `null` here is slightly misleading